.github/workflows/compile-test.yml

```diff
@@ -0,0 +1,72 @@
+name: Compile Test
+
+on:
+  push:
+
+jobs:
+  compile-headers:
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: ["ubuntu-latest", "macos-latest"]
+        language: ["objective-c", "objective-c++"]
+        cflags: ["", "-fobjc-arc"]
+    runs-on: ${{ matrix.runner }}
+    steps:
+      - name: Checkout theos/sdks
+        uses: actions/checkout@v3
+        with:
+          repository: theos/sdks
+          ref: bb425abf3acae8eac328b828628b82df544d2774 # pinned, should be updated as appropriate
+          path: sdks
+
+      - name: Checkout theos/headers
+        uses: actions/checkout@v3
+        with:
+          path: headers
+
+      - name: Compile All
+        run: |
+          UNCHECKED_DIRECTORIES=(
+              "IOKit"
+              "openssl"
+          )
+          UNCHECKED_FILES=(
+              "QuartzCore/QuartzCore-Structs.h" # known to not work
+              "rocketbootstrap/rocketbootstrap_dynamic.h" # noted to not include directly
+              "libundirect/libundirect_hookoverwrite.h" # user's responsibility to include either the base or dynamic header
+          )
+
+          HEADER_DIR="headers"
+          # cut to drop the leading directory
+          for HEADER in $(find "${HEADER_DIR}" -name "*.h" | cut -c"$(( ${#HEADER_DIR} + 2))"-); do
+            IS_UNCHECKED="false"
+            
+            for UNCHECKED_DIRECTORY in "${UNCHECKED_DIRECTORIES[@]}"; do
+                if [[ "${HEADER}" == "${UNCHECKED_DIRECTORY}"/* ]]; then
+                        IS_UNCHECKED="true"
+                        break
+                fi
+            done
+            for UNCHECKED_FILE in "${UNCHECKED_FILES[@]}"; do
+                if [[ "${HEADER}" == "${UNCHECKED_FILE}" ]]; then
+                        IS_UNCHECKED="true"
+                        break
+                fi
+            done
+            if [[ "${IS_UNCHECKED}" == "true" ]]; then
+                continue
+            fi
```

Comment:
This looks great! Just wanted to note that we may be able to do away with this check and the corresponding var if the breaks in the prior two if statements are changed to `continue 2`.   

```
~$ continue --help
continue: continue [n]
    Resume for, while, or until loops.

    Resumes the next iteration of the enclosing FOR, WHILE or UNTIL loop.
    If N is specified, resumes the Nth enclosing loop.

    Exit Status:
    The exit status is 0 unless N is not greater than or equal to 1.
```

Reply:
Thank you! This is great, simplifies the code overall.